openhcl/underhill_attestation/src/lib.rs

```diff
@@ -51,16 +53,32 @@ use zerocopy::FromZeroes;
 /// An attestation error.
 #[derive(Debug, Error)]
 #[error(transparent)]
-pub struct Error(ErrorInner);
+pub struct Error(AttestationErrorInner);
 
-impl<T: Into<ErrorInner>> From<T> for Error {
+impl Error {
+    fn log_error_id<T: Any + Debug>(e: &T) {
+        let value_any = e as &dyn Any;
+        let type_name = std::any::type_name::<T>();
+
+        match value_any.downcast_ref::<String>() {
+            Some(as_string) => {
+                tracing::error!(CVM_ALLOWED, "Unexpected string {:?}", as_string);
```

Comment:
I think this creates additional confusion / noise.

Reply:
Code removed.  This is a API usage error, it should probably be an assert.